stdlib/socket.pyi

```diff
@@ -1,509 +1,385 @@
 import sys
+from _typeshed import ReadableBuffer, WriteableBuffer
+from collections.abc import Iterable
 from enum import IntEnum, IntFlag
-from typing import Any, BinaryIO, Iterable, List, Optional, TextIO, Tuple, TypeVar, Union, overload
+from io import RawIOBase
+from typing import Any, BinaryIO, Optional, TextIO, TypeVar, Union, overload
 from typing_extensions import Literal
 
-# ----- Constants -----
-# Some socket families are listed in the "Socket families" section of the docs,
-# but not the "Constants" section. These are listed at the end of the list of
-# constants.
-#
-# Besides those and the first few constants listed, the constants are listed in
-# documentation order.
+import _socket
+from _socket import (
```

Comment:
Why can't we just do `from _socket import *`? That's what happens at runtime, and it's easier to maintain than keeping that huge list of constants in sync.

Reply:
I tried this at first. See my frustrated comment about mypy not handling overrides of star imports correctly above.

Reply:
I added a comment to the import that explains this decision.